src/children/Category.hack

```diff
@@ -0,0 +1,27 @@
+/*
+ *  Copyright (c) 2004-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the MIT license found in the
+ *  LICENSE file in the root directory of this source tree.
+ *
+ */
+
+namespace Facebook\XHP\ChildValidation;
+
+use namespace HH\Lib\Str;
+
+final class Category extends LeafConstraint {
+  public function __construct(private string $category) {
+  }
+
+  public function legacySerializeAsLeaf(): (LegacyConstraintType, string) {
+    return tuple(
+      LegacyConstraintType::CATEGORY,
+      $this->category
+        |> Str\strip_prefix($$, '%')
+        |> Str\replace($$, ':', '__')
+        |> Str\replace($$, '-', '_'),
```

Comment:
Perhaps reuse the existing "xhpize class name" method.

Reply:
Intentionally not doing that, as `xhp class` syntax will remove all XHP class name mangling, but I'm not expecting it to change category mangling (oh hi @johanoskarsson :) )